### PR TITLE
Add contributing guidelines and format every js file

### PR DESCRIPTION
**Pull request by EXtremeExploit**
literally title, now with contributing guidelines i think there should be some rules to follow or to at least have A BIT of code quality check. still a work in progress as i still have to add the workflow that checks for each push and actually format all files acording to .eslintrc. @Tanza3D @mulraf if you guys feel like adding any comments on what rules should follow i'll gladly accept them. there are some that are impossible like functions or variables not being used as they may be used in other files, using of undefined variables has to be added manually to the `globals` section of the .eslintrc, i think it also helps to keep track of every global so its fine. thankfully there aren't that many globals (i think)

same goes for the contrib guidelines please i want feedback

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+* Update translations on crowding
+* Report bugs by opening issues here on github
+* 
+
+### I do know how to code and i want to contribute with code :D
+
+## Pull Requests
+* Destination branch should be `main`
+* Source branch **MUST NOT** be `main`
+* Must pass all tests
```
Comment (minusQuantumNeko) [inline CONTRIBUTING.md:14]:
```suggestion
* **Must** pass all [checks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks)!
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+### If you don't want to or not know how to contribute code:
+* Update translations on crowding
+* Report bugs by opening issues here on github
+* 
+
+### I do know how to code and i want to contribute with code :D
+
+## Pull Requests
+* Destination branch should be `main`
+* Source branch **MUST NOT** be `main`
```
Comment (minusQuantumNeko) [inline CONTRIBUTING.md:13]:
```suggestion
* Source branch **MUST __NOT__** be `main`!
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+### If you don't want to or not know how to contribute code:
+* Update translations on crowding
+* Report bugs by opening issues here on github
+* 
+
+### I do know how to code and i want to contribute with code :D
+
+## Pull Requests
+* Destination branch should be `main`
```
Comment (minusQuantumNeko) [inline CONTRIBUTING.md:12]:
```suggestion
* Destination branch **must** be `main`!
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+* Source branch **MUST NOT** be `main`
+* Must pass all tests
+
+## Code
+
+#### JS
+* All JS code must be formatted according to `.eslintrc` rules.
```
Comment (minusQuantumNeko) [inline CONTRIBUTING.md:19]:
```suggestion
* All **`JS`** code **must** be formatted according to `.eslintrc` rules.
```

**Review by minusQuantumNeko — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# CONTRIBUTING GUIDELINES
+### So you want to contribute, great!
+
+### If you don't want to or not know how to contribute code:
```
[the remark this answers is not in the document]
Comment (Tanza3D) [reply CONTRIBUTING.md:4]: i think this is more an option for people who also don't know how to code, a better choice might be:
`If you don't want to, or do not know how to code or contribute your code`

[the remark this answers is not in the document]
Comment (EXtremeExploit) [reply CONTRIBUTING.md:4]: i mean not know how to contribute code as if you don't know how to code, i guess i can reword this as `I don't feel confident enough in my experience in coding D: What can i do?`?

[the remark this answers is not in the document]
Comment (minusQuantumNeko) [reply CONTRIBUTING.md:4]: > i think this is more an option for people who also don't know how to code, a better choice might be: `If you don't want to, or do not know how to code or contribute your code`

Updated suggestion

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+* Report bugs by opening issues here on github
+* 
+
+### I do know how to code and i want to contribute with code :D
+
+## Pull Requests
+* Destination branch should be `main`
+* Source branch **MUST NOT** be `main`
+* Must pass all tests
+
```
[the remark this answers is not in the document]
Comment (EXtremeExploit) [reply CONTRIBUTING.md:15]: honestly i have mixed feelings about this one, i agree that it is good to have it, but only make use of it for stuff like fixing stuff right before merge that is impossible to have the guy making the pr have any objections about it or if the PR is dead but easy to pickup. I don't know about you guys but at least thats how i see it, i do still think that code changes made this way have to be reviewed by the guy making the pr too. i don't know how to put this in the guidelines or if you guys agree. just my opinion

[the remark this answers is not in the document]
Comment (Tanza3D) [reply CONTRIBUTING.md:15]: i don't see the point in having this honestly

[the remark this answers is not in the document]
Comment (minusQuantumNeko) [reply CONTRIBUTING.md:15]: Honestly, it doesn't harm to have that, better to have that when we need it than not. Also even osu-wiki says that if I remember correctly

[the remark this answers is not in the document]
Comment (Tanza3D) [reply CONTRIBUTING.md:15]: i guessss

[the remark this answers is not in the document]
Comment (EXtremeExploit) [reply CONTRIBUTING.md:15]: now that i think about it we can put it as recommended :thinking: . just a suggestion

[the remark this answers is not in the document]
Comment (minusQuantumNeko) [reply CONTRIBUTING.md:15]: But what does it change, lol?

**Comment by Tanza3D** [comment]
is css class name guidelines doable?
